pkg/services/validator/maas.go

```diff
@@ -0,0 +1,661 @@
+	Each availability zone should have no more than 1 rule configured.
+	`)
+
+	validateResources, err := prompts.ReadBool("Enable Resource Availability validation", true)
+	if err != nil {
+		return err
+	}
+
+	if !validateResources {
+		c.Validator.ResourceAvailabilityRules = nil
+		return nil
+	}
+
+	for i, r := range c.Validator.ResourceAvailabilityRules {
+		r := r
+		if err := readMaasResourceRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+
+	addRules := true
+	if c.Validator.ResourceAvailabilityRules == nil {
+		c.Validator.ResourceAvailabilityRules = make([]vpmaasapi.ResourceAvailabilityRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another Resource Availability rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readMaasResourceRule(c, &vpmaasapi.ResourceAvailabilityRule{}, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add another Resource Availability rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+// nolint:dupl
+func readMaasResourceRule(c *components.MaasPluginConfig, r *vpmaasapi.ResourceAvailabilityRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &vpmaasapi.ResourceAvailabilityRule{
+			Name:      "",
+			AZ:        "",
+			Resources: []vpmaasapi.Resource{},
+		}
+	}
+
+	err := initMaasRule(r, "Resource Availability", ruleNames)
+	if err != nil {
+		return err
+	}
+
+	if r.Name == "" {
+		name, err := prompts.ReadText("Rule Name", "", false, -1)
+		if err != nil {
+			return err
+		}
+		r.Name = name
+	}
+
+	if r.AZ == "" {
+		az, err := prompts.ReadText("Availability Zone", "az1", false, -1)
+		if err != nil {
+			return err
+		}
+		r.AZ = az
+	}
+
+	addResources := true
+
+	for addResources {
+		resource, err := readMaasResource()
+		if err != nil {
+			return err
+		}
+
+		r.Resources = append(r.Resources, resource)
+		addResources, err = prompts.ReadBool("Add another resource", false)
+		if err != nil {
+			return err
+		}
+	}
+	if idx == -1 {
+		c.Validator.ResourceAvailabilityRules = append(c.Validator.ResourceAvailabilityRules, *r)
+	} else {
+		c.Validator.ResourceAvailabilityRules[idx] = *r
+	}
+	return nil
+}
+
+// nolint:dupl
+func readMaasResource() (vpmaasapi.Resource, error) {
+	res := vpmaasapi.Resource{}
+
+	numMachines, err := prompts.ReadInt("Minimum number of machines", "1", 1, -1)
+	if err != nil {
+		return res, err
+	}
+	res.NumMachines = numMachines
+
+	numCPU, err := prompts.ReadInt("Minimum CPU cores per machine", "4", 1, -1)
+	if err != nil {
+		return res, err
+	}
+	res.NumCPU = numCPU
+
+	ram, err := prompts.ReadInt("Minimum RAM in GB", "16", 1, -1)
+	if err != nil {
+		return res, err
+	}
+	res.RAM = ram
+
+	disk, err := prompts.ReadInt("Minimum Disk capacity in GB", "256", 1, -1)
+	if err != nil {
+		return res, err
+	}
+	res.Disk = disk
+
+	pool, err := prompts.ReadText("Machine pool", "", true, -1)
```

Comment:
Would be awesome if you could list available pools and provide a selection here.